smailyforprestashop.php

```diff
@@ -304,12 +336,61 @@ public function getContent()
                 'smailyforprestashop',
                 'SmailyCartCron'
             ),
+            'smaily_rss_available_category_ids' => $this->recursivelyNormalizeCategoriesForTemplate($categories),
+            'smaily_rss_selected_category_id' => pSQL(Configuration::get('SMAILY_RSS_CATEGORY_ID')),
+            'smaily_rss_limit' => pSQL(Configuration::get('SMAILY_RSS_LIMIT')),
+            'smaily_rss_order_by' => pSQL(Configuration::get('SMAILY_RSS_ORDER_BY')),
+            'smaily_rss_order_way' => pSQL(Configuration::get('SMAILY_RSS_ORDER_WAY')),
             )
         );
         // Display settings form.
         return $output .= $this->display(__FILE__, 'views/templates/admin/smaily_configure.tpl');
     }
 
+    /**
+     * Recursively go through categories in array and normalize for template.
+     *
+     * @param array $categories Enabled categories in Prestashop catalog.
+     *
+     * @return array Categories in format: array(category id => category name).
+     */
+    private function recursivelyNormalizeCategoriesForTemplate($categories)
+    {
+        $normalized = array();
+        foreach ( $categories as $category ) {
+            $normalized[$category['id_category']] = $category['name'];
+            if (isset($category['children']) && is_array($category['children'])) {
+                $normalized += $this->recursivelyNormalizeCategoriesForTemplate($category['children']);
+            }
+        }
+        return $normalized;
+    }
+
+    /**
+     * Make CRON URL with query parameters.
+     *
+     * @return string $url
+     * e.g example.com/en/module/smailyforprestashop/SmailyRssFeed?limit=50&order_by=date_upd&order_way=desc&id_category=2
+     */
+    private function generateCronUrlFromSettings()
```

Comment:
Method name does not represent the functionality - method is building a RSS feed URL, not a CRON URL.

Reply:
I don't know why I did this. Added